test/e2e/tests/metrics/attribution/marketing-cookieid.spec.ts

```diff
@@ -0,0 +1,226 @@
+ * indicator to our data team that the shape of data will change.
+ *
+ * @param mockServer
+ * @returns
+ */
+const mockSegment = async (mockServer: Mockttp) => {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Main Menu Opened' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+};
+
+describe('Marketing cookieId', function (this: Suite) {
+  it('update state when the script is executed and the metrics context is updated with the cookieId when an event is fired', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        dappPaths: ['./tests/metrics/attribution/mock-page'],
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+            dataCollectionForMarketing: true,
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockSegment,
+      },
+      async ({
+        driver,
+        mockedEndpoint: mockedEndpoints,
+      }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+
+        await driver.openNewPage(`http://127.0.0.1:8080`);
+        // wait for state to update
+        await driver.delay(5000);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        const uiState = await getCleanAppState(driver);
+        assert.equal(uiState.metamask.marketingCampaignCookieId, 12345);
+
+        await driver.clickElement(selectors.accountOptionsMenuButton);
+        const events = await getEventPayloads(
+          driver,
+          mockedEndpoints as MockedEndpoint[],
+        );
+        assert.equal(events.length, 1);
+        const eventContext = events[0].context;
+        assert.equal(eventContext.marketingCampaignCookieId, 12345);
+      },
+    );
+  });
+  it('doesnot update state when dataCollectionForMarketing is already false ', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        dappPaths: ['./tests/metrics/attribution/mock-page'],
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockSegment,
+      },
+      async ({
+        driver,
+        mockedEndpoint: mockedEndpoints,
+      }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+
+        await driver.openNewPage(`http://127.0.0.1:8080`);
+        // wait for state to update
+        await driver.delay(5000);
```

Comment:
Question: would 3 seconds enough ?

Reply:
The test was flaky, so QA suggested keeping a longer delay.

Reply:
Personally i believe we should find a solution for flakyness before putting long delay, as it is not the fix if the memory is enough given to each test to run in parallel. Especially if this is introduced the first time and becomes flaky already. After checking the logic, i believe this can be fixed starting from changes as 
```
 await driver.switchToWindowWithTitle(
          WINDOW_TITLES.ExtensionInFullScreenView,
        );
await openDapp(driver);
```
Do you think this is a more valid solution as extension stream would be ready to listen to messages, then we post the message while opening the dapp?

Reply:
When we finish the ' unlockWallet () ', the extension will be loaded in full-screen view. I believe `driver.switchToWindowWithTitle` is used only to switch between tabs/windows.

Reply:
oh yes that's true/ my bad